replit_river/session.py

```diff
@@ -228,9 +237,15 @@ async def _check_to_close_session(self) -> None:
             await asyncio.sleep(
                 self._transport_options.close_session_check_interval_ms / 1000
             )
+            if self._state != SessionState.ACTIVE:
+                # already closing
+                return
+            # calculate the value now before comparing it so that there are no
```

Comment:
Trying to understand this better, Why would this be different before or after
```
if not self._close_session_after_time_secs:
  continue
```?

Reply:
ah we don't want to have any `await` in the middle of doing a check / operation on the member field

Reply:
Ah I see your point, that make sense!
Thanks for the clarification, out of curiousity, how did you find this bug?